### PR TITLE
Upgrade transformers library - [ENG 85]

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- Allow transformers library versions between 4.13.0 to 4.26.0

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -1,10 +1,16 @@
+# This file is automatically @generated by Poetry and should not be changed by hand.
```
Comment (ancalita) [inline poetry.lock:1]: Out of curiosity, which poetry version are you using locally @varunshankar ? Haven't seen this comment before, neither the parameter `files` which now shows up for every package.

File: pyproject.toml [hunk 2]
```diff
@@ -262,7 +262,7 @@ version = "~3.2.0"
 optional = true
 
 [tool.poetry.dependencies.transformers]
-version = "~4.13.0"
+version = ">=4.13.0, <=4.26.0"
```
Comment (ancalita) [inline pyproject.toml:265]: Based on this [comment](https://rasahq.atlassian.net/browse/ENG-85?focusedCommentId=16951), this should be a strict `<4.26.0` rather.

**Review by ancalita — COMMENTED** [review]
Thanks for moving this over to `main` 💯 
Left a question about the constraint for dependency specification.

File: poetry.lock (hunk 1 above)
Comment (varunshankar) [reply poetry.lock:1]: Got this info from `poetry about`

Poetry - Package Management for Python

Version: 1.3.1
Poetry-Core Version: 1.4.0

Comment (ancalita) [reply poetry.lock:1]: Oh I see, can you please downgrade to `1.2.2`, that's the version we use in the CI and for building Docker images at the moment.

File: pyproject.toml (hunk 2 above)
Comment (varunshankar) [reply pyproject.toml:265]: Thomas suggested me to keep it `<=4.26.0`. And I also verified that nothing breaks in `4.26.0`

Comment (ancalita) [reply pyproject.toml:265]: Sounds good, just thought of double-checking.

File: poetry.lock (hunk 1 above)
Comment (varunshankar) [reply poetry.lock]: Ya, I have downgraded poetry now to `1.2.2` and created a new lock file

**Review by ancalita — APPROVED** [review]

**Pull request merged**